packages/js/components/src/date-time-picker-control/date-time-picker-control.tsx

```diff
@@ -88,40 +94,76 @@ export const DateTimePickerControl: React.FC< DateTimePickerControlProps > = ( {
 		);
 	}
 
-	const onChangeCallback = useCallback(
-		( newInputString: string ) => {
-			if ( ! isMounted.current ) return;
-
-			const newDateTime = parseMoment( newInputString );
-			const isValid = newDateTime.isValid();
-
-			if ( isValid ) {
-				setLastValidDate( newDateTime );
-			}
-
-			if ( onChange ) {
-				onChange(
-					isValid ? formatMomentIso( newDateTime ) : newInputString,
-					isValid
-				);
-			}
-		},
-		[ onChange ]
-	);
+	// We setup the debounced handling of the input string changes using
+	// useRef because useCallback does *not* guarantee that the resulting
+	// callback function will not be recreated, even if the dependencies
+	// haven't changed (this is because of it's use of useMemo under the
+	// hood, which also makes not guarantee). And, even if it did, the
+	// equality check for useCallback dependencies is by reference. So, if
+	// the "same" function is passed in, but it is a different instance, it
+	// will trigger the recreation of the callback.
+	//
+	// With useDebounce, if the callback function changes, the current
```

Comment:
If `useDebounce` does not satisfy partialy/fully our requirements why we are using it. A lot of code was written just to make that function works. I think we can implement a custom `useDebounce` with your improvements (whitout consider @wordpress/compose) that help us all reusing it. Or re-touch the original one with your improvements.

Reply:
That's a fair point! I'll think about this!

Reply:
Would you be okay with this being a follow up PR to introduce a custom `useDebounce`? Alternatively, I can create a separate PR ahead of this one to add in a custom `useDebounce`. I'm thinking a follow up might be a good compromise since this PR does improve the current behavior and has tests around it. That way, the follow up can focus just on a custom `useDebounce`, and we have this component and the tests to help us ensure the custom `useDebounce` works as intended (of course, it will have its own tests as well).